### PR TITLE
initial version of Quarkus.io image

### PR DESCRIPTION
**Pull request by vorburger**
see https://github.com/tqvarnst/quarkus-workshop/issues/2

### CODE REVIEW

**Comment by vorburger** [comment]
@cescoffier @benoitf @l0rd @dharmit would one of you be able to review and merge this? Tx!

What do I then have to do to get it built and published on https://registry.centos.org/che-stacks/centos-quarkus.io-jdk8?

**Comment by benoitf** [comment]
cc @eivantsov

**Comment by benoitf** [comment]
@vorburger if https://github.com/eclipse/che-dockerfiles/blob/master/build.sh is handling the format of your new directory, it will be built

**Comment by dneary** [comment]
Thank you @vorburger!

**Comment by vorburger** [comment]
> What do I then have to do to get it built and published
> @vorburger if [build.sh](https://github.com/eclipse/che-dockerfiles/blob/master/build.sh) is handling the format of your new directory, it will be built

Oh, right; got it. FTR: I just locally ran this `build.sh` with my minor #241 extension, and see that it will.

**Comment by l0rd** [comment]
@vorburger I have a couple of comments:

1) I may miss some context here but if you are going to use the new sidecar workspaces (with Theia) you should not need to have a Dockerfile here. In other words you could just reuse the official quarkus image in your workspace/stack. Or, if you need some extra dependencies on top of the official quarkus image, your Dockerfile could live in the quarkus-quickstarts repository. It would be easier to manage it there and your Dockerfile lifecycle is closely tied to the quickstart (i.e. when you change the quickstart source code you want the image to be updated right away). If instead you are going to use the good old wsagent based workspaces you need indeed your image to be based on the `che-stacks/centos-jdk8` and it's lifecycle should be closely tied to the base image lifecycle so your PR is perfectly fine and you can ignore my comment.

2) We used to need a [.cccp.yml](https://github.com/eclipse/che-dockerfiles/blob/master/recipes/centos_go/.cccp.yml) file to have images published in registry.centos.org. Has that changed @benoitf ?

**Comment by vorburger** [comment]
@l0rd the context here is https://github.com/tqvarnst/quarkus-workshop/ and specifically https://github.com/tqvarnst/quarkus-workshop/issues/2. This PR proposes (something like) what https://github.com/tqvarnst/quarkus-workshop/commit/c2c5c3c7ad39bb9528113193a978260d63969c35 does locally in that project but as an upstream in Che itself. IMHO it will make future maintenance easier to have this here, together with the other very similar containers maintained here?

Full disclosure: I would need to learn more about the new sidecar workspaces (with Theia) VS the good old wsagent based workspaces - my current quite limited understanding makes me think that https://github.com/tqvarnst/quarkus-workshop/ stlll uses that old approach. 

> We used to need a .cccp.yml file to have images published in registry.centos.org.

OK, added in 8ef0f3c.

**Comment by l0rd** [comment]
To get more info about the new workspace model https://che.eclipse.org/the-new-superpowers-of-che-workspaces-243967a2010

As a consequence of this change, instead of having to maintain the build and runtime images in this repository we can leverage external ones. For instance the [new Java stack](https://github.com/eclipse/che/blob/master/ide/che-core-ide-stacks/src/main/resources/stacks.json#L658) uses the official [maven image from dockerhub](https://github.com/eclipse/che/blob/master/ide/che-core-ide-stacks/src/main/resources/stacks.json#L719): we do not need to maintain our patched image anymore. 

This new approach scales really well and makes it easy for us to add stacks (gradle, tomcat, spring etc...) with a minimal effort. 

In other words we need to have a discussion amongst Che maintainers to decide if this repository (`che-dockerfiles`) should still exist with the advent of workspace.next.

**Comment by vorburger** [comment]
> In other words we need to have a discussion amongst Che maintainers to decide if this repository (`che-dockerfiles`) should still exist with the advent of workspace.next.

OK, then I'll just wait for that. Sorry I'm not super u-to-date, but in the mean time, isn't this repo what currently is still used? Just based on the git log, this repo seems to still be active? We sure could put something like this PR (plus https://github.com/redhat-developer/che-dockerfiles/pull/54; or https://github.com/tqvarnst/quarkus-workshop/tree/master/che-stack) into https://github.com/quarkusio/quarkus-images, but then we would have to copy/paste some of the scripts used - I'm not sure if that is really such a good idea longer term?

**Comment by l0rd** [comment]
@vorburger I am ok to merge this PR but as far as I know you need to add a `.cccp.yml` file: https://github.com/centos/container-index#cccpyaml

Once we have merged this one I would like to add a Quarkus.io sidecar based stack as well. We can guide you do do so if you are interested or we can take care of it.

**Comment by vorburger** [comment]
> you need to add a .cccp.yml file

added in 8ef0f3c - OK?

> Once we have merged this one I would like to add a Quarkus.io sidecar based stack as well. We can guide you do do so if you are interested or we can take care of it.

OK. I'll reach out to you when this is merged to understand what exactly is needed for that.

**Comment by ibuziuk** [comment]
@l0rd @eivantsov it looks like there is still no image pushed to the Pulling repository registry.centos.org/che-stacks/centos-quarkus.io-jdk8 even though it is added to the container index - https://github.com/CentOS/container-index/pull/395/files
I suspect we need to trigger the build somehow

**Comment by vorburger** [comment]
So `docker pull registry.centos.org/che-stacks/centos-quarkus-io-jdk8:latest` seems to work now (as in it pulls, haven't actually tested the image, yet). I can't find it on the UI though, and have opened https://github.com/CentOS/container-pipeline-service/issues/689 about that.

**Comment by ibuziuk** [comment]
@vorburger hmm, was it just fixed ?

downstream CI seems to be still failing - https://ci.centos.org/job/devtools-eclipse-che-build-dockerfiles/618/console

> Step 1/18 : FROM registry.centos.org/che-stacks/centos-quarkus.io-jdk8
> Trying to pull repository registry.centos.org/che-stacks/centos-quarkus.io-jdk8 ... 
> Pulling repository registry.centos.org/che-stacks/centos-quarkus.io-jdk8
> Error: image che-stacks/centos-quarkus.io-jdk8:latest not found
> Connection to 172.19.2.134 closed.

**Comment by vorburger** [comment]
> @vorburger hmm, was it just fixed ?
> downstream CI seems to be still failing

The build you linked ran 17h ago, and I tried 30 min ago... but that's not it, I just figured that there is a minor typo diff!! Note `registry.centos.org/che-stacks/centos-quarkus.io-jdk8` (NOK) VS `registry.centos.org/che-stacks/centos-quarkus-io-jdk8` (OK, with the dot between quarkus and io replaced by a dash!)

BTW note new https://github.com/redhat-developer/rh-che/issues/1393, to wrap this up...
